### PR TITLE
added TouchSelection class to select a phrase from a set of grids/pages

### PR DESCRIPTION
**Pull request by jaspalsinghsaini**
Added TouchSelection class whose object would take as input an array of phrases; and this object would populate the grid structure with a subset of those phrases along with the functional buttons. The functional button 'Next' can be used to view the next set of phrases from the same array for selection.

### CODE REVIEW

File: SwitchSpeak-iOS/SwitchSpeak-iOS/CrumbManager.swift [hunk 1]
```diff
@@ -60,10 +60,23 @@ class CrumbStack {
...
+		if (self.getString() != "") {	//	i.e. the stack is not empty
+			items.removeLast().label.removeFromSuperview()
+		}
+		
     }
+	
+	/*
+	*	remove all the breadcrumbs
+	*/
+	func emptyCrumbStack() {
```
Comment (jaspalsinghsaini) [inline SwitchSpeak-iOS/SwitchSpeak-iOS/CrumbManager.swift]: let me know if you would recommend me some other method for emptying the stack

File: SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/TouchSelection.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		index = (index + gridSize - 4) % phrases.count
+	}
+	
+	/*
+	*	This function selects the currently highlighted phrase in the grid, and it returns nil is currently
+	*	a set of hrases are highlighted in the grid
+	*	This function also refills the grid with the next set of phrases incase the 'Next' phrase is selected
+	*	a more suitable function name can be considered since its name is the same as a function in the TouchGrid class
+	*/
+	func makeSelection() -> String? {
```
Comment (jaspalsinghsaini) [inline SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/TouchSelection.swift]: this function name matches another function name in TouchGrid case. Couldn't think of any other suitable name....any suggestions?

File: SwitchSpeak-iOS/SwitchSpeak-iOS/ViewController.swift [hunk 3]
```diff
@@ -10,27 +10,62 @@ import UIKit
...
 	@IBOutlet weak var TapButton: UIButton!
 	var breadcrumbs = CrumbStack()
-    var touchGrid:TouchGrid?
+    var touchSelection:TouchSelection?
 	
 	override func viewDidLoad() {
 		super.viewDidLoad()
-        
+		//	dummy array of phrases for the first grid selection
+		let dummyPhrases: [String] = ["1", "2", "3", "4", "5", "6", "7", "8","9","10","11"]
```
Comment (jaspalsinghsaini) [inline SwitchSpeak-iOS/SwitchSpeak-iOS/ViewController.swift]: there are a couple of such dummy arrays....we can update these once the database is set up.

**Comment by rgerd** [comment]
Hey I'm looking through the code and it looks great. Just two small things I noticed right off the bat are typos in comments (check your new code in TouchSelection and ViewController), and that we're going with opening brackets on the same line as the function declaration:
`func foo() {
    ...
}`

**Comment by rgerd** [comment]
I also have a design suggestion:
- make the ViewController's CrumbStack another public instance variable of TouchSelection, right next to `touchGrid:TouchGrid?`. 
- Take [this massive thing](https://github.com/rgerd/SwitchSpeak/blob/2e07e0529874aad40f31bffda98eae64c888be74/SwitchSpeak-iOS/SwitchSpeak-iOS/ViewController.swift#L36-L69), create a file called ActionButton.swift in the TouchGrid folder, define a class called `ButtonAction`, and move the logic under each `case` into their own static functions in `ButtonAction` (i.e. [`doOops`](https://github.com/rgerd/SwitchSpeak/blob/2e07e0529874aad40f31bffda98eae64c888be74/SwitchSpeak-iOS/SwitchSpeak-iOS/ViewController.swift#L37-L41), [`doHome`](https://github.com/rgerd/SwitchSpeak/blob/2e07e0529874aad40f31bffda98eae64c888be74/SwitchSpeak-iOS/SwitchSpeak-iOS/ViewController.swift#L43-L46), etc.). These each take a `TouchSelection` as their only parameter. Since `TouchSelection` now has the `CrumbStack` and `TouchGrid`, these functions should have everything they need. (See [`GlobalSettings`](https://github.com/rgerd/SwitchSpeak/blob/master/SwitchSpeak-iOS/SwitchSpeak-iOS/Settings/GlobalSettings.swift) for an example of a class with only static functions).
- Create an `enum` of `Strings` outside of the class but inside the ActionButton.swift file (like I do in [`UserSettings`](https://github.com/rgerd/SwitchSpeak/blob/master/SwitchSpeak-iOS/SwitchSpeak-iOS/Settings/UserSettings.swift)) called ActionButton that holds the titles of the action buttons:
```swift
enum ActionButton : String {
    case oops = "Oops"
    case home = "Home"
    case next = "Next"
    ...
}
```
- Make a static function in `ButtonAction` called `callAction` that takes an `ActionButton` and does the original switch you had in the `ViewController`, but now over the cases in `ActionButton`, kind of like what we do [here](https://github.com/rgerd/SwitchSpeak/blob/master/SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/TreeFactory.swift#L73-L80), and call the "`do`" functions described in the second point in this list.
- Now for the fancy part. In the `makeSelection` function in `TouchSelection`, still use [this line](https://github.com/rgerd/SwitchSpeak/pull/13/files#diff-ae2b30aaad141017ace492e6a8a07652R64); add the check [if `choice` is `nil` return](https://github.com/rgerd/SwitchSpeak/blob/master/SwitchSpeak-iOS/SwitchSpeak-iOS/ViewController.swift#L28-L30). Then [use this](https://stackoverflow.com/a/32820588/6421697). I'm talking about using the line starting with `guard` to either convert `choice` into an `ActionButton`, in which case you call `ButtonAction.callAction` with that `ActionButton` version of `choice`; `else` you run the logic you originally wrote [here](https://github.com/rgerd/SwitchSpeak/blob/2e07e0529874aad40f31bffda98eae64c888be74/SwitchSpeak-iOS/SwitchSpeak-iOS/ViewController.swift#L56-L67).
- Remove the return in `TouchSelection.makeSelection` and make it void.
- When everything is said and done, the [`TapButton`]() function should look like this:
```swift
@IBAction func TapButton(_ sender: Any) {
    touchSelection!.makeSelection()
}
```
And everything should work the same way, but now we have a central place where action button titles and methods are cleanly defined and can be called succinctly from anywhere.

A design methodology we want to follow is keeping all of the ViewControllers as short as possible, and spreading things out into separate files as much as possible. This makes git merge errors less common (since we won't all be editing the same files), and the view controller can be spared for very specific things like button touches and simple UI setup.

**Comment by jaspalsinghsaini** [comment]
I have refactored the code present the UIViewController into another ActionButton class. Please take a look at it and let me know your thoughts.

**Pull request merged**